lib/auth/utils.ts

```diff
@@ -63,30 +63,27 @@ export const getDashboardLoginURL = (callbackUrl: string) => {
  *   if loggedIn
  *     console.log('Token is valid!')
  */
-export const loginIfTokenValid = async (token: string) => {
+export const loginIfTokenValid = async (token: string): Promise<boolean> => {
 	if (_.isEmpty(token?.trim())) {
 		return false;
 	}
 	const balena = getBalenaSdk();
 
-	return balena.auth
-		.getToken()
-		.catchReturn(undefined)
-		.then((currentToken) =>
-			balena.auth
-				.loginWithToken(token)
-				.return(token)
-				.then(balena.auth.isLoggedIn)
-				.tap((isLoggedIn) => {
-					if (isLoggedIn) {
-						return;
-					}
+	let currentToken;
+	try {
+		currentToken = await balena.auth.getToken();
```

Comment:
It can't be moved there because
```ts
await balena.auth.loginWithToken(token);
```
will switch to the new token and the "current"/previous token will be lost at that point